dir.props

```diff
@@ -12,6 +12,7 @@
   <PropertyGroup>
     <OSGroup Condition="'$(OSGroup)'==''">$(__BuildOS)</OSGroup>
     <Platform Condition="'$(Platform)'==''">$(__BuildArch)</Platform>
+    <HostOS Condition="'$(HostOS)'==''">$(__HostOS)</HostOS>
```

Comment:
`HostOS` in msbuild world is just `OS`. msbuild has `HostOS` built-in property that describes the host OS. I do not think we need HostOS property for msbuild.

Reply:
msbuild $(OS) is set to 'Unix' though on OSX. I cannot use OSGroup either, since that is based off of __BuildOS, which gets overridden to be 'WebAssembly'. If I use $(OS), how do I distinguish between OSX and Linux?

Reply:
I can see TargetOS gets initialized like this in Microsoft.NETCore.Native.targets:
<TargetOS Condition="'$(TargetOS)' == '' and '$(OS)' == 'Unix' and Exists('/Applications')">OSX</TargetOS>

But this prop isn't imported in the WebAssembly.csproj.
Would it be cleaner to copy this property definition over?

Right now having the logic in multiple places seems fragile (the different nix distros are differentiated between in the build scripts so I figured OSX detection should be the same place?)

Reply:
The clean way to do this would be to let the build system pick the right assets from the NuGet package. Similar to how it is done e.g. for RyuJIT: https://github.com/dotnet/corert/blob/master/src/ILCompiler/RyuJIT/RyuJIT.depproj .

As a hack, I think the HostOS property should be fine. I cannot think of anything better.